src/task_queue/index.ts

```diff
@@ -87,12 +89,25 @@ export class TaskInfo extends AsyncResource implements Task {
 
       this.filename = filename;
       this.name = name;
+      // TODO: This should not be global
```

Comment:
And it should a UUID to avoid overflow.

Reply:
Interesting, do you see a need for universal uniqueness?

My first idea was to provide a simple counter that limits to the SMI optimization and re-starts; maybe in combination with the Worker ID; and if need for enabling more complex use-cases ask the caller to provide an ID generator

Reply:
High rate tasks submission workload could overflow it if no restart is done since months. Isn't the task execution counting already done via histogram?
Can it also overflow?

Reply:
True, but do we would need to track down every task individually for more than 30 days? I imagine that maybe you'll like it for debugging purposes, but not sure how much after that.

For that, caller should implement a custom solution that fits its use case; I'd like to provide a good defaults with less overhead and allow extensibility.

>Isn't the task execution counting already done via histogram?
Can it also overflow?

Noup. I imagine the task id just for debugging purposes, beyond that I don't see much of a use for it.